### PR TITLE
Added object detection pickup option, will likely turn into a while held button for teleop later

### PR DESCRIPTION
**Pull request by Coolearthsky**

### CODE REVIEW

File: comp/swerve100/src/main/java/org/team100/frc2024/RobotContainer.java [hunk 1]
```diff
@@ -171,11 +171,8 @@ public RobotContainer(TimedRobot robot) throws IOException {
 
     
 
-        VisionDataProvider visionDataProvider = new VisionDataProvider(
-                m_layout,
-                poseEstimator,
-                poseEstimator::getEstimatedPosition);
-        visionDataProvider.enable();
+        NotePosition24ArrayListener notePositionDetector = new NotePosition24ArrayListener();
```
Comment (truher) [inline comp/swerve100/src/main/java/org/team100/frc2024/RobotContainer.java:174]: you shouldn't remove the vision data provider.

File: lib/src/main/java/org/team100/lib/motion/drivetrain/manual/DriveWithNoteRotation.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ * 
+ * The twist components, x, y, and theta, are mapped directly to the
+ * corresponding ChassisSpeeds components (and scaled).
+ */
+public class DriveWithNoteRotation {
+    private final PIDController m_pidController;
+    private final Telemetry t = Telemetry.get();
+    private final SwerveKinodynamics m_swerveKinodynamics;
+    private final String m_name;
+    private final NotePosition24ArrayListener notePosition;
```
Comment (truher) [inline lib/src/main/java/org/team100/lib/motion/drivetrain/manual/DriveWithNoteRotation.java:24]: you should follow the style here: member variable names start with "m_"

File: lib/src/main/java/org/team100/lib/motion/drivetrain/manual/DriveWithNoteRotation.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    public DriveWithNoteRotation(String parent, SwerveKinodynamics swerveKinodynamics, PIDController pidController, NotePosition24ArrayListener arrayListener) {
+        notePosition = arrayListener;
```
Comment (truher) [inline lib/src/main/java/org/team100/lib/motion/drivetrain/manual/DriveWithNoteRotation.java:26]: the name in the constructor argument list should be the same as the field name, except for the "m_"

File: lib/src/main/java/org/team100/lib/motion/drivetrain/manual/DriveWithNoteRotation.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+import org.team100.lib.telemetry.Telemetry;
+import org.team100.lib.telemetry.Telemetry.Level;
+import org.team100.lib.util.DriveUtil;
+import org.team100.lib.util.Names;
+
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.geometry.Twist2d;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+/**
+ * Transform manual input into ChassisSpeeds.
```
Comment (truher) [inline lib/src/main/java/org/team100/lib/motion/drivetrain/manual/DriveWithNoteRotation.java:14]: you should make this comment describe what this class actually does

File: lib/src/main/java/org/team100/lib/motion/drivetrain/manual/DriveWithNoteRotation.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    public DriveWithNoteRotation(String parent, SwerveKinodynamics swerveKinodynamics, PIDController pidController, NotePosition24ArrayListener arrayListener) {
+        notePosition = arrayListener; 
+        m_pidController = pidController;
+        m_swerveKinodynamics = swerveKinodynamics;
+        m_name = Names.append(parent, this);
+    }
+
+    /**
+     * Clips the input to the unit circle, scales to maximum (not simultaneously
+     * feasible) speeds, and then desaturates to a feasible holonomic velocity.
```
Comment (truher) [inline lib/src/main/java/org/team100/lib/motion/drivetrain/manual/DriveWithNoteRotation.java:34]: make this comment describe what the code actually does

File: comp/vision/gamepiece_finder24.py [hunk 6]
```diff
@@ -47,7 +47,7 @@ def initialize_nt(self):
         """Start NetworkTables with Rio as server, set up publisher."""
         self.inst = NetworkTableInstance.getDefault()
         self.inst.startClient4("gamepiece_finder24")
-        # this is always the RIO IP address; set a matching static IP on your
+        # this is always the RIO IP address; set sa matching static IP on your
```
Comment (truher) [inline comp/vision/gamepiece_finder24.py:50]: typo here?

**Review by truher — CHANGES_REQUESTED** [review]
looks pretty good, i left a few comments.

File: lib/src/main/java/org/team100/lib/motion/drivetrain/manual/DriveWithNoteRotation.java (hunk 5 above)
Comment (Coolearthsky) [reply lib/src/main/java/org/team100/lib/motion/drivetrain/manual/DriveWithNoteRotation.java:35]: This was fixed

File: comp/swerve100/src/main/java/org/team100/frc2024/RobotContainer.java (hunk 1 above)
Comment (Coolearthsky) [reply comp/swerve100/src/main/java/org/team100/frc2024/RobotContainer.java:180]: Also fixed

**Review by truher — COMMENTED** [review]
looks great.  i can't wait to see this thing in action.

**Pull request merged**